command/certificate/lint.go

```diff
@@ -93,30 +94,17 @@ func lintAction(ctx *cli.Context) error {
 	}
 
 	var (
-		crtFile = ctx.Args().Get(0)
-		block   *pem.Block
+		crtFile  = ctx.Args().Get(0)
+		roots    = ctx.String("roots")
+		insecure = ctx.Bool("insecure")
+		block    *pem.Block
 	)
-	if strings.HasPrefix(crtFile, "https://") {
-		var (
-			err     error
-			rootCAs *x509.CertPool
-		)
-		if roots := ctx.String("roots"); roots != "" {
-			rootCAs, err = x509util.ReadCertPool(roots)
-			if err != nil {
-				errors.Wrapf(err, "failure to load root certificate pool from input path '%s'", roots)
-			}
-		}
-		addr := strings.TrimPrefix(crtFile, "https://")
-		if !strings.Contains(addr, ":") {
-			addr += ":443"
-		}
-		conn, err := tls.Dial("tcp", addr, &tls.Config{RootCAs: rootCAs})
+	if isURL, _, addr := trimURLPrefix(crtFile); isURL {
+		peerCertificates, err := getPeerCertificates(addr, roots, insecure)
 		if err != nil {
-			return errors.Wrapf(err, "failed to connect")
+			return errors.WithStack(err)
```

Comment:
if getPeerCertificates returns an error from the `pkg/errors` package it already have a stack, there's no need to replace it as we might drop part of the stack.